pkg/apis/messaging/v1/subscription_validation.go

```diff
@@ -21,17 +21,29 @@ import (
 
 	"github.com/google/go-cmp/cmp/cmpopts"
 	"k8s.io/apimachinery/pkg/api/equality"
+	"knative.dev/eventing/pkg/apis/feature"
+	cn "knative.dev/eventing/pkg/crossnamespace"
 	"knative.dev/pkg/apis"
 	duckv1 "knative.dev/pkg/apis/duck/v1"
 	"knative.dev/pkg/kmp"
 )
 
 func (s *Subscription) Validate(ctx context.Context) *apis.FieldError {
+	// s.Validate(ctx) because krshaped is defined on the entire subscription, not just the spec
 	errs := s.Spec.Validate(ctx).ViaField("spec")
 	if apis.IsInUpdate(ctx) {
 		original := apis.GetBaseline(ctx).(*Subscription)
 		errs = errs.Also(s.CheckImmutableFields(ctx, original))
 	}
+	if feature.FromContext(ctx).IsEnabled(feature.CrossNamespaceEventLinks) {
+		crossNamespaceError := cn.CheckNamespace(ctx, s)
+		if crossNamespaceError != nil {
+			errs = errs.Also(crossNamespaceError)
+		}
+	}
+	// validate whether or not the permissions are all correct
+	// check if interface works
+	// validate namespace
```

Comment:
why is this comment useful ? what does it add that is not clear from the code itself ?